### PR TITLE
Add JSONStrings formats

### PR DESCRIPTION
**Pull request by hczhcz**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add JSONStrings formats which output data in arrays of strings

Detailed description / Documentation draft:

> ## JSONStrings {#jsonstrings}
>
> Differs from JSON and JSONCompact only in that data rows are output in arrays of strings.

The purpose is to get readable table data (like those in `Pretty` format) in restful services.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
JSONCompactStrings maybe?

File: src/DataTypes/DataTypeNullable.cpp [hunk 1]
```diff
@@ -308,15 +308,22 @@ ReturnType DataTypeNullable::deserializeTextQuoted(IColumn & column, ReadBuffer
...
 
 
 void DataTypeNullable::deserializeWholeText(IColumn & column, ReadBuffer & istr, const FormatSettings & settings) const
 {
     safeDeserialize(column, *nested_data_type,
-        [&istr] { return checkStringByFirstCharacterAndAssertTheRestCaseInsensitive("NULL", istr); },
+        [&istr]
+        {
+            return checkStringByFirstCharacterAndAssertTheRestCaseInsensitive("NULL", istr)
+                || checkStringByFirstCharacterAndAssertTheRest("ᴺᵁᴸᴸ", istr);
```
Comment (nikitamikhaylov) [inline src/DataTypes/DataTypeNullable.cpp:325]: Do you have a test on it?

File: src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp [hunk 2]
```diff
@@ -138,10 +143,25 @@ void JSONEachRowRowInputFormat::readField(size_t index, MutableColumns & columns
     {
         seen_columns[index] = read_columns[index] = true;
         const auto & type = getPort().getHeader().getByPosition(index).type;
-        if (format_settings.null_as_default && !type->isNullable())
-            read_columns[index] = DataTypeNullable::deserializeTextJSON(*columns[index], in, format_settings, type);
+
+        if (yield_strings)
+        {
+            // notice: null_as_default on "null" strings is not supported
```
Comment (nikitamikhaylov) [inline src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:149]: Why? Is it really difficult? I think it just like code below.

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
There is no new integration tests and feature is isolated, so lets merge.

**Pull request merged**